src/mame/nichibutsu/galivan.cpp

```diff
@@ -272,12 +273,28 @@ class ninjemak_state : public galivan_state
 	void blit_trigger_w(uint8_t data);
 	void vblank_ack_w(uint8_t data);
 	void gfxbank_w(uint8_t data);
-	TILE_GET_INFO_MEMBER(get_bg_tile_info);
-	TILE_GET_INFO_MEMBER(get_tx_tile_info);
+	virtual TILE_GET_INFO_MEMBER(get_bg_tile_info) override;
+	virtual TILE_GET_INFO_MEMBER(get_tx_tile_info) override;
 	void palette(palette_device &palette) const;
 	uint32_t screen_update(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect);
 };
 
+class ninjemakt_state : public galivan_state
+{
+public:
+	ninjemakt_state(const machine_config &mconfig, device_type type, const char *tag) :
+		galivan_state(mconfig, type, tag)
+	{ }
+
+protected:
+	virtual void video_start() override ATTR_COLD;
+
+private:
+	virtual TILE_GET_INFO_MEMBER(get_tx_tile_info) override;
+
+};
```

Comment:
Why have you made the tile info functions `virtual` when every class that wants a different implementation overrides `video_start` to create the tilemaps itself, and hence will pick up the implementation from the most derived class anyway?  It doesn’t need to go through the vtable at all.